dev-java/gson/gson-2.11.0.ebuild

```diff
@@ -0,0 +1,84 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="com.google.code.gson:gson:${PV}"
+JAVA_TESTING_FRAMEWORKS="junit-4"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="Gson JSON library"
+HOMEPAGE="https://github.com/google/gson"
+SRC_URI="https://github.com/google/gson/archive/gson-parent-${PV}.tar.gz
+	test? ( https://repo1.maven.org/maven2/com/google/truth/truth/1.1.3/truth-1.1.3.jar )"
+S="${WORKDIR}/gson-gson-parent-${PV}/gson"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+
+DEPEND="
+	dev-java/error-prone-annotations:0
+	>=virtual/jdk-17:*
+	test? (
+		dev-java/guava:0
+		dev-java/guava-testlib:0
+	)"
+# src/test/java/com/google/gson/functional/Java17RecordTest.java:78:
+# error: records are not supported in -source 11
+RDEPEND=">=virtual/jre-17:*"
+
+DOCS=(
+	../CHANGELOG.md
+	../GsonDesignDocument.md
+	../README.md
+	../Troubleshooting.md
+	../UserGuide.md
+)
+
+PATCHES=( "${FILESDIR}/gson-2.10.2-GsonVersionDiagnosticsTest.patch" )
+
+JAVA_CLASSPATH_EXTRA="error-prone-annotations"
+JAVA_RESOURCE_DIRS="src/main/resources"
+JAVA_SRC_DIR=(
+	"src/main/java"
+	"src/main/java-templates"
+)
+
+JAVA_TEST_EXCLUDES=(
+	# requires the test class to be obfuscated using proguard which we do not have atm
+	"com.google.gson.functional.EnumWithObfuscatedTest"
+	# testComGoogleGsonAnnotationsPackage(com.google.gson.regression.OSGiTest)
+	# junit.framework.AssertionFailedError: Cannot find com.google.gson OSGi bundle manifest
```

Comment:
i guess this means we are not putting osgi bundle manifest in the `META-INF/MANIFEST.MF` file, right? if so, we could fix this.

Reply:
maybe. i don't even know what parts (lines) belong to `osgi bundle manifest`. upstream seems to build it from a `bnd.bnd` file. replicating this in the ebuild would require much better understanding of how to use `bnd`. it's not getting clear for me by reading `mvn's` log.